public/api/v1/overview.php

```diff
@@ -0,0 +1,738 @@
+            if (array_key_exists($checker,
+                $dynamic_analysis_data[$i][$build_group['name']])) {
+                $found = true;
+                break;
+            }
+        }
+        if (!$found) {
+            continue;
+        }
+
+        $group_response = array();
+        $group_response['name'] = $build_group['name'];
+        $group_response['name_clean'] =
+            sanitize_string($build_group['name']);
+
+        $chart_data = get_DA_chart_data($build_group['name'], $checker);
+        $group_response['chart'] = $chart_data;
+
+        $value = get_current_DA_value($build_group['name'], $checker);
+        $group_response['value'] = $value;
+        $groups_response[] = $group_response;
+    }
+    $DA_response['groups'] = $groups_response;
+    $dynamic_analyses_response[] = $DA_response;
+}
+$response['dynamicanalyses'] = $dynamic_analyses_response;
+
+// static analysis
+$static_analyses_response = array();
+foreach ($static_groups as $static_group) {
+    // Skip this group if no data was found for it.
+    $found = false;
+    for ($i = 0; $i < $date_range; $i++) {
+        $static_array = &$overview_data[$i][$static_group['name']];
+        foreach ($static_measurements as $measurement) {
+            if (array_key_exists($measurement, $static_array)) {
+                $found = true;
+                break;
+            }
+        }
+        if ($found) {
+            break;
+        }
+    }
+    if (!$found) {
+        continue;
+    }
+
+    $SA_response = array();
+    $SA_response['group_name'] = $static_group['name'];
+    $SA_response['group_name_clean'] = sanitize_string($static_group['name']);
+    $measurements_response = array();
+    foreach ($static_measurements as $measurement) {
+        $measurement_response = array();
+        $measurement_response['name'] = $measurement;
+        $measurement_response['nice_name'] = sanitize_string($measurement);
+        $measurement_response['sort'] = $sort["build_$measurement"];
+        $value = get_current_value($static_group['name'], $measurement);
+        $measurement_response['value'] = $value;
+
+        $chart_data = get_chart_data($static_group['name'], $measurement);
+        $measurement_response['chart'] = $chart_data;
+        $measurements_response[] = $measurement_response;
+    }
+    $SA_response['measurements'] = $measurements_response;
+    $static_analyses_response[] = $SA_response;
+}
+$response['staticanalyses'] = $static_analyses_response;
+
+$end = microtime_float();
+$response['generationtime'] = round($end - $start, 3);
+echo json_encode(cast_data_for_JSON($response));
+
+// Replace various characters that trip up Javascript with underscores.
+function sanitize_string($input_string)
+{
+    $retval = str_replace(' ', '_', $input_string);
+    $retval = str_replace('-', '_', $retval);
+    $retval = str_replace('.', '_', $retval);
+    $retval = str_replace('(', '_', $retval);
+    $retval = str_replace(')', '_', $retval);
```

Comment:
This sanitization might be less important now that we're using Angular directives instead of XSLT's code injection.  I agree with your recommendation to collapse this down into a single call to ``str_replace``.